contracts/modules/TransferManager/VolumeRestrictionTM.sol

```diff
@@ -0,0 +1,934 @@
+     */
+    function _individualRestrictionCheck(
+        address _from,
+        uint256 _amount,
+        bool _isTransfer
+    )
+        internal
+        returns(Result)
+    {   
+        uint256 fromTimestamp;
+        uint256 sumOfLastPeriod = 0;
+        if (bucketToUser[_from].timestamps.length == 0) {
+            // It will execute when the txn is performed first time after the addition of individual restriction
+            fromTimestamp = individualRestriction[_from].startTime;
+        } else {
+            // Picking up the last timestamp
+            fromTimestamp = bucketToUser[_from].timestamps[bucketToUser[_from].timestamps.length -1];
+        }
+        // Calculating the difference of days
+        uint256 diffDays = BokkyPooBahsDateTimeLibrary.diffDays(fromTimestamp, now); 
+        uint256[] memory passedTimestamps = new uint256[](diffDays);
+        // Check with the bucket and parse all the new timestamps to calculate the sumOfLastPeriod
+        // re-using the local variables to avoid the stack too deep error.
+        (sumOfLastPeriod, fromTimestamp, passedTimestamps) = _bucketCheck(
+            fromTimestamp,
+            diffDays,
+            _from,
+            bucketToUser[_from],
+            false,
+            individualRestriction[_from]
+        );
+        if (_checkValidAmountToTransact(sumOfLastPeriod, _amount, individualRestriction[_from])) {
+            if (_isTransfer) {
+                _updateIndividualStorage(
+                    passedTimestamps,
+                    _from,
+                    fromTimestamp,
+                    _amount,
+                    diffDays,
+                    sumOfLastPeriod
+                );
+            }
+            return Result.NA;
+        } 
+        return Result.INVALID;  
+    }
+
+
+    function _globalTxCheck(
+        uint256 _sumOfLastPeriod,
+        uint256 _dailyAmount,
+        uint256 _amount
+    ) 
+        internal
+        view
+        returns(bool)
+    {
+        // Checking whether the global day restriction is added or not if yes then calculate
+        // the total amount get traded on a particular day (~ _fromTime)
+        if (!_checkValidAmountToTransact(_dailyAmount, _amount, dailyGlobalRestriction) &&
+                now <= dailyGlobalRestriction.endTime && now >= dailyGlobalRestriction.startTime
+        ) {
+            return false;
+        }
+        return _checkValidAmountToTransact(_sumOfLastPeriod, _amount, globalRestriction);
+    }
+
+    function _checkValidAmountToTransact(
+        uint256 _sumOfLastPeriod,
+        uint256 _amountToTransact,
+        VolumeRestriction _restriction
+    ) 
+        internal
+        view
+        returns (bool)
+    {    
+        uint256 _allowedAmount = 0;
+        if (_restriction.typeOfRestriction == RestrictionType.Variable) {
+            _allowedAmount = (_restriction.allowedPercentageOfTokens.mul(ISecurityToken(securityToken).totalSupply()))/ 10 ** 18;
```

Comment:
```suggestion
            _allowedAmount = (_restriction.allowedPercentageOfTokens.mul(ISecurityToken(securityToken).totalSupply()))/ uint256(10) ** 18;
```

Without explicit casting, It might have unexpected behaviour.